data/desktop/Makefile

```diff
@@ -0,0 +1,38 @@
+#
+# Copyright (C) 2018 Canonical Ltd
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License version 3 as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+BINDIR = /usr/bin
+DATADIR = /usr/share
+
+SOURCES = snap-userd-autostart.desktop.in
+DESKTOP_FILES = $(SOURCES:.in=)
+
+.PHONY: all
+all: $(DESKTOP_FILES)
+
+.PHONY: install
+install: $(DESKTOP_FILES)
+	# NOTE: old (e.g. 14.04) GNU coreutils doesn't -D with -t
+	install -d -m 0755 $(DESTDIR)/$(DATADIR)/applications
+	install -m 0644 -t $(DESTDIR)/$(SYSTEMDSYSTEMUNITDIR) $^
```

Comment:
Sorry if I'm being dense, but this seems like we are installing a desktop file to the SYSTEMDSYSTEMUNITDIR. Shouldn't we instead be installing a unit file to the systemd user session directory? What about systems that don't use a systemd user session? I was trying to figure out how 'snapd userd --autostart' was called in the user's session and didn't see how that was happening.

Reply:
Since @bboozzoo and @mvo5 are offline, I am going to add the Blocked tag to make sure that https://github.com/snapcore/snapd/pull/4942#discussion_r179836777 is answered. Feel free to remove the tag when satisfied on how `snapd userd --autostart` is going to be launched. (I did it this way rather than 'Request Changes' so you aren't blocked on me)

Reply:
Thanks for catching this. Fixing in another patch